indico/web/client/js/react/components/ListFilter.jsx

```diff
@@ -89,38 +108,58 @@ export default function ListFilter({
     return null;
   };
 
-  const setSearchText = value => {
-    if (onChangeSearchText) {
-      onChangeSearchText(value);
-      return;
+  const setSearchText = useCallback(
+    value => {
+      if (onChangeSearchText) {
+        onChangeSearchText(value);
+        return;
+      }
+      setInternalSearchText(value);
+      value = value.toLowerCase().trim();
+      const filtered = list.filter(e => matchFilters(filters, e) && matchSearch(value, e));
+      onChangeList(new Set(filtered.map(e => e.id)));
+    },
+    [onChangeSearchText, onChangeList, list, matchFilters, matchSearch, filters]
+  );
+
+  const toggleFilter = (filterKey, optionValue) => {
+    const optionObject = filterOptions
+      .find(x => x.key === filterKey)
+      .options.find(x => x.value === optionValue);
```

Comment:
I think we can avoid some extra looping
```suggestion
    const optionObject = optionsMap.get(filterKey).options.find(x => x.value === optionValue);
```